contracts/StakeManager.sol

```diff
@@ -229,63 +274,39 @@ contract StakeManager is Ownable {
         if (_acceptMigration) {
             accounts[_vault] = _account;
         } else {
-            multiplierSupply -= _account.multiplier;
-            stakeSupply -= _account.balance;
+            totalSupplyMP -= _account.currentMP;
+            totalSupplyBalance -= _account.balance;
         }
     }
+
     /**
      * @dev Only callable from old manager.
      * @notice Increase total MP from old manager
      * @param _increasedMP amount MP increased on account after migration initialized
      */
-
     function increaseMPFromMigration(uint256 _increasedMP) external onlyOldManager {
-        multiplierSupply += _increasedMP;
-    }
-
-    function calcMaxMultiplierIncrease(
-        uint256 _increasedMultiplier,
-        uint256 _currentMp,
-        uint256 _lockUntil,
-        uint256 _stake
-    )
-        private
-        view
-        returns (uint256 _maxToIncrease)
-    {
-        uint256 newMp = _increasedMultiplier + _currentMp;
-        if (block.timestamp > _lockUntil) {
-            //not locked, limit to max_boost
-            return newMp > _stake * MAX_BOOST ? _stake * MAX_BOOST - _currentMp : _increasedMultiplier;
-        } else {
-            // locked, ignore cap
-            return _increasedMultiplier;
-        }
-    }
-
-    function processEpoch() private {
-        if (block.timestamp >= epochEnd() && address(migration) == address(0)) {
-            //finalize current epoch
-            epochs[currentEpoch].epochReward = epochReward();
-            epochs[currentEpoch].totalSupply = totalSupply();
-            pendingReward += epochs[currentEpoch].epochReward;
-            //create new epoch
-            currentEpoch++;
-            epochs[currentEpoch].startTime = block.timestamp;
-        }
+        totalSupplyMP += _increasedMP;
     }
 
+    /**
+     * @notice Process account until limit has reached
+     * @param account Account to process
+     * @param _limitEpoch Until what epoch it should be executed
+     */
     function processAccount(Account storage account, uint256 _limitEpoch) private {
-        processEpoch();
+        if (account.lockUntil == 0) {
+            revert StakeManager__AccountNotInitialized();
```

Comment:
This check is happening in multiple places.

How do you feel about introducing a `onlyInitializedAccounts` modifier?

Reply:
Will do that.